statsmodels/tools/data.py

```diff
@@ -121,3 +122,16 @@ def _is_recarray(data):
         return isinstance(data, np.core.recarray)
     else:
         return isinstance(data, np.rec.recarray)
+
+def _as_array_with_name(obj):
+    """
+    Call np.asarray() on obj and attempt to get the name if its a Series.
+
+    Returns
+    -------
+    array_and_name: tuple[np.ndarray, str | None]
+        The data casted to np.ndarra and the series name or None
+    """
+    if is_series(obj):
+        return (np.asarray(obj), obj.name)
+    return (np.asarray(obj), None)
```

Comment:
we need a default name her

Reply:
Just added. I thought it was probably a good idea to make the caller provide a default so there's always a meaningful name vs. `""`.